lib/combine_pdf/renderer.rb

```diff
@@ -107,7 +107,7 @@ def format_hash_to_pdf(object)
         end
       end
       # remove extra page references.
-      object[:Contents].delete(is_reference_only: true, referenced_object: { indirect_reference_id: 0, raw_stream_content: '' }) if object[:Type] == :Page
+      object[:Contents].delete(is_reference_only: true, referenced_object: { indirect_reference_id: 0, raw_stream_content: '' }) if object[:Type] == :Page && object[:Contents]
```

Comment:
I have tested
In my test `object[:Contents]` is always a `Hash`, not an `Array`.  Why you suggest `Array` is it's assumed state?

Why is the code `object[:Contents].delete(is_reference_only: true, referenced_object: { indirect_reference_id: 0, raw_stream_content: '' }) if object[:Type] == :Page` necessary?
A `Hash` is not a valid parameter of the `delete` method. See: https://ruby-doc.org/core-2.3.3/Hash.html#method-i-delete. 

Removing te line gives the same result?